processor/src/decoder/block_stack.rs

```diff
@@ -177,6 +156,35 @@ impl BlockInfo {
     }
 }
 
+// BLOCK CONTEXT INFO
+// ================================================================================================
+
+/// Contains information about an execution context. Execution context are relevant only for CALL
+/// blocks.
+#[derive(Debug, Default, Clone, Copy)]
+pub struct ExecutionContextInfo {
+    /// Context ID of the block's parent.
+    pub parent_ctx: u32,
+    /// Value of free memory pointer right before a CALL instruction is executed.
+    pub fmp: Felt,
+    /// Depth of the operand stack right before a CALL operation is executed.
+    pub stack_depth: u32,
+    /// Address of the top row in the overflow table right before a CALL operations is executed.
+    pub next_overflow_addr: Felt,
+}
```

Comment:
I did this mostly because it felt weird having every single field in a struct prefixed with the same thing, and also, I could kind of justify saying that, something like `ctx` is a property of the parent (i.e., parent block always has a unique context ID), while something like `fmp` is fluid and does not really describe the parent (it is more like `fmp` value in the parent context right before the call).

But I think the above is a very weak argument, and it seems like my choice caused enough confusion. So, I just added `parent_` prefix to all fields.